File: 0645-set-mismatch/0645-set-mismatch.py
class Solution(object):
    def findErrorNums(self, nums):
        duplicate = -1
        missing = -1
        for num in nums:
            index = abs(num)-1
            if nums[index]<0:
                duplicate = abs(num)
            else:nums[index] = -nums[index]
        for i in range(len(nums)):
            if nums[i]>0:
                missing = i+1
        return [duplicate,missing]

        # count=Counter(nums)
        # ans=[]
        # for key,value in count.items():
        #     if value>1:
        #         ans.append(key)
        #         break
        # for x in range(1,len(nums)+1):
        #     if x not in count.keys():
        #         ans.append(x)
        #         break
        # return ans     

File: 0645-set-mismatch/0645-set-mismatch.java
class Solution 
{
    public int[] findErrorNums(int[] nums) 
    {
        int j=0;
        while(j<nums.length)
        {
            int correct = nums[j]-1;
            if(nums[j] != nums[correct])
                swap(nums, j, correct);
            else
                j++;
        }
        for(int i=0;i<nums.length;i++)
        {
            if(nums[i] != i+1)
                return new int[]{nums[i],i+1};
        }
        return new int[]{-1,-1};
    }
    public void swap(int[] nums,int first,int second)
    {
        int temp = nums[first];
        nums[first] = nums[second];
        nums[second] = temp;
    }
}